crates/modelardb_manager/src/metadata.rs

```diff
@@ -13,188 +13,478 @@
         };
 
-        // Create the necessary tables in the metadata database.
+        // Create the necessary tables in the metadata Delta Lake.
         metadata_manager
-            .create_manager_metadata_database_tables()
+            .create_manager_metadata_delta_lake_tables()
             .await?;
 
         Ok(metadata_manager)
     }
 
     /// If they do not already exist, create the tables that are specific to the manager metadata
-    /// database.
-    /// * The manager_metadata table contains metadata for the manager itself. It is assumed that
+    /// Delta Lake.
+    /// * The `manager_metadata` table contains metadata for the manager itself. It is assumed that
     ///   this table will only have a single row since there can only be a single manager.
-    /// * The nodes table contains metadata for each node that is controlled by the manager.
-    ///   If the tables exist or were created, return [`Ok`], otherwise return [`sqlx::Error`].
-    async fn create_manager_metadata_database_tables(&self) -> Result<(), sqlx::Error> {
-        let mut transaction = self.metadata_database_pool.begin().await?;
-
+    /// * The `nodes` table contains metadata for each node that is controlled by the manager.
+    ///
+    /// If the tables exist or were created, return [`Ok`], otherwise return [`DeltaTableError`].
+    async fn create_manager_metadata_delta_lake_tables(&self) -> Result<(), DeltaTableError> {
         // Create the manager_metadata table if it does not exist.
-        sqlx::query("CREATE TABLE IF NOT EXISTS manager_metadata (key TEXT PRIMARY KEY)")
-            .execute(&mut *transaction)
+        self.metadata_delta_lake
+            .create_delta_lake_table(
+                "manager_metadata",
+                vec![StructField::new("key", DataType::STRING, false)],
+            )
             .await?;
 
         // Create the nodes table if it does not exist.
-        sqlx::query(
-            "CREATE TABLE IF NOT EXISTS nodes (
-                 url TEXT PRIMARY KEY,
-                 mode TEXT NOT NULL
-             )",
-        )
-        .execute(&mut *transaction)
-        .await?;
-
-        transaction.commit().await?;
+        self.metadata_delta_lake
+            .create_delta_lake_table(
+                "nodes",
+                vec![
+                    StructField::new("url", DataType::STRING, false),
+                    StructField::new("mode", DataType::STRING, false),
+                ],
+            )
+            .await?;
 
         Ok(())
     }
 
-    /// Retrieve the key for the manager from the manager_metadata table. If a key does not already
-    /// exist, create one and save it to the database.
-    pub async fn manager_key(&self) -> Result<Uuid, sqlx::Error> {
-        let maybe_row = sqlx::query("SELECT key FROM manager_metadata")
-            .fetch_optional(&self.metadata_database_pool)
+    /// Retrieve the key for the manager from the `manager_metadata` table. If a key does not
```

Comment:
That is a good point. However, I have not changed anything yet since I think the authentication and security changes later will heavily deprecate this design anyway.